### PR TITLE
add changes for haxe4 compatibility

### PR DESCRIPTION
**Pull request by jdonaldson**
Here's a bunch of changes related to haxe4 compatibility and best practices.  One big breaking change is that Lambda now returns `Array`instead of `List`.  I just went ahead and refactored it, but if you want to maintain a haxe3 compat, it's probably better to use the hx3compat library.

There's only a few critical issues, the remainder are minor formatting/syntax changes that will remove some new warnings.  I'll call out some of the bigger code changes below.

### CODE REVIEW

File: compiler/fbs/Converter.hx [hunk 1]
```diff
@@ -71,18 +68,17 @@ class Converter {
 		});
 
 		var printer:haxe.macro.Printer = new haxe.macro.Printer();
-		currentModule.types.map(function(t) {
+        for (t in currentModule.types){
```
Comment (jdonaldson) [inline compiler/fbs/Converter.hx:71]: The Lambda.map functions changed (to require a return).  This expression works fine as a for-loop.

File: compiler/fbs/Converter.hx [hunk 2]
```diff
@@ -71,18 +68,17 @@ class Converter {
...
 
 		return currentModule;
 	}
 
 	function storeDeclTypes(parsedObj:ParsedObject):Void {
 		for(field in Reflect.fields(parsedObj)) {
-			Reflect.field(parsedObj, field);
-			(cast Reflect.field(parsedObj, field):Array<Dynamic>).map(function(decl:FbsDeclaration) {
-				switch decl {
+            for (decl in cast ( Reflect.field(parsedObj, field), Array<Dynamic>)) {
```
Comment (jdonaldson) [inline compiler/fbs/Converter.hx:80]: The Lambda.map functions changed (to require a return).  This expression works fine as a for-loop.

File: compiler/fbs/Converter.hx [hunk 3]
```diff
@@ -117,7 +114,7 @@ class Converter {
 
 	function convertEnum(decl:FbsDeclaration):TypeDefinition {
 		var enumObj:FbsEnum = decl.getParameters()[0];
-		var fields:List<Field> = enumObj.ctors.mapi(function(i:Int, ctor:FbsEnumCtor):Field {
+		var fields:Array<Field> = enumObj.ctors.mapi(function(i:Int, ctor:FbsEnumCtor):Field {
```
Comment (jdonaldson) [inline compiler/fbs/Converter.hx:117]: The Lambda functions now return `Array` instead of list.

File: compiler/fbs/Converter.hx [hunk 4]
```diff
@@ -601,27 +595,27 @@ class Converter {
 			access: [APublic, AStatic],
 			pos: nullPos
 		}
-		
-		var funcAddFields:List<Array<Field>> = structObj.fields.mapi(function(i:Int, field:FbsTableField) {
+
+		var funcAddFields:Array<Array<Field>> = structObj.fields.mapi(function(i:Int, field:FbsTableField) {
```
Comment (jdonaldson) [inline compiler/fbs/Converter.hx:599]: Another List->Array change from Lambda.

File: compiler/fbs/Converter.hx [hunk 5]
```diff
@@ -30,7 +27,7 @@ typedef StructPadding = {
 }
 
 typedef FieldType = {
-	type:ComplexType, 
+	type:ComplexType,
```
Comment (jdonaldson) [inline compiler/fbs/Converter.hx:30]: I automatically strip trailing whitespace.

File: compiler/fbs/Converter.hx [hunk 6]
```diff
@@ -154,7 +151,7 @@ class Converter {
 			return structObj.fields.map(function(field:FbsStructField) {
 				var fieldType:FieldType;
 				switch field.type {
-					case TPrimitive(t): 
+					case TPrimitive(_):
```
Comment (jdonaldson) [inline compiler/fbs/Converter.hx:154]: Haxe4 language server complains about unused variable (for `t`).  Replacing this variable with `_` will remove the warning.

**Review by Zentendo — APPROVED** [review]
Approved.

**Review by troyedwardsjr — APPROVED** [review]

**Pull request merged**